src/lib/draw/route/RouteMode.svelte

```diff
@@ -61,11 +68,58 @@
       changeMode
     );
   });
+
+  async function fetchWithProgress(url, setProgress: (number) => void) {
+    const response = await fetch(url);
+    const reader = response.body.getReader();
+
+    const contentLength = parseInt(response.headers.get("Content-Length"));
+
+    let receivedLength = 0;
+    let chunks = [];
+    while (true) {
+      const { done, value } = await reader.read();
+      if (done) {
+        break;
+      }
+
+      chunks.push(value);
+      receivedLength += value.length;
+
+      const percent = (100.0 * receivedLength) / contentLength;
+      setProgress(percent);
+    }
+
+    let allChunks = new Uint8Array(receivedLength);
+    let position = 0;
+    for (let chunk of chunks) {
+      allChunks.set(chunk, position);
+      position += chunk.length;
+    }
+
+    return allChunks;
+  }
+
+  function routeToolInitialised() {
+    progress = 100;
+    routeToolReady = true;
+  }
 </script>
 
-{#if !routeTool}
-  <!-- TODO the text should be fixed, and the progress bar float -->
-  <div bind:this={progress}>Route tool loading...</div>
+{#if !routeToolReady && !failedToLoadRouteTool && !downloadComplete}
+  <label for="route-loading">Route tool loading</label>
+  <progress id="route-loading" value={progress} />
+{:else if downloadComplete && !routeToolReady && !failedToLoadRouteTool}
+  <label for="route-unpacking">Route data unpacking</label>
+  <progress id="route-unpacking" />
```

Comment:
I really like the default animation of bouncing back and forth!

So the strategy is:
1) Show a 0-100 progress bar while we're downloading, up to the Content-Length of the .gzip file
2) We'll exceed 100%, because the browser is unpacking and passing along more than that Content-Length (and we have no way of knowing the uncompressed size, unless we store and send back a second header somehow)
3) As soon as we hit 100%, we switch to the bouncey animation and say unpacking. We might still be loading from the network for some of this, but it's fine.